src/renderer/components/cluster-manager/lens-views.ts

```diff
@@ -1,62 +1,37 @@
-import { ipcRenderer, WebviewTag } from "electron";
-import { observable, when } from "mobx";
+import { observable } from "mobx";
 import { ClusterId, clusterStore } from "../../../common/cluster-store";
-import { clusterIpc } from "../../../common/cluster-ipc";
-import { clusterViewURL, getMatchedCluster, getMatchedClusterId } from "./cluster-view.route"
-import { navigate } from "../../navigation";
+import { getMatchedCluster } from "./cluster-view.route"
 import logger from "../../../main/logger";
 
 export interface LensView {
   isLoaded?: boolean
   clusterId: ClusterId;
-  view: WebviewTag
+  view: HTMLIFrameElement
 }
 
 export const lensViews = observable.map<ClusterId, LensView>();
 
-export async function navigateInClusterView(path: string, clusterId: ClusterId) {
-  // select active cluster in common view
-  if (clusterId !== getMatchedClusterId()) {
-    clusterStore.setActive(clusterId);
-    navigate(clusterViewURL({ params: { clusterId } }));
-  }
-  // navigate in cluster-view when ready
-  await when(() => hasLoadedView(clusterId))
-  ipcRenderer.sendTo(getViewId(clusterId), "menu:navigate", path);
-}
-
 export function hasLoadedView(clusterId: ClusterId): boolean {
   return !!lensViews.get(clusterId)?.isLoaded;
 }
 
-export function getViewId(clusterId: ClusterId): number {
-  const webview = lensViews.get(clusterId)?.view
-  if (webview) {
-    return webview.getWebContentsId()
-  }
-}
-
-// todo: figure out how to replace <webview>-tag to <iframe> with nodeIntegration=true
-export function initView(clusterId: ClusterId) {
+export async function initView(clusterId: ClusterId) {
   if (!clusterId || lensViews.has(clusterId)) {
     return;
   }
-  logger.info(`[CLUSTER-VIEW]: init dashboard, clusterId=${clusterId}`)
-  const parentElem = document.getElementById("lens-views"); // defined in cluster-manager's css-grid
-  const webview = document.createElement("webview");
-  webview.setAttribute("src", `//${clusterId}.${location.host}`)
-  webview.setAttribute("nodeintegration", "true")
-  webview.setAttribute("enableremotemodule", "true")
-  webview.addEventListener("did-finish-load", () => {
-    logger.info(`[CLUSTER-VIEW]: loaded, clusterId=${clusterId}`)
-    clusterIpc.init.invokeFromRenderer(clusterId); // push cluster-state to webview and init render
+  logger.info(`[LENS-VIEW]: init dashboard, clusterId=${clusterId}`)
```

Comment:
Is it not possible to create `iframe`'s using normal React components?

Reply:
I think we need to create iframes manually so that we avoid any possible DOM manipulation from React. If DOM changes iframe is reloaded -> state is lost.

Reply:
Ah fair, good point